integrations/nvidia/src/haystack_integrations/utils/nvidia/utils.py

```diff
@@ -105,22 +102,13 @@ def validate_hosted_model(class_name: str, model_name: str, client: Any) -> None
         if not model.client:
             warn_msg = f"Unable to determine validity of {model.id}"
             warnings.warn(warn_msg, stacklevel=1)
-        if model.model_type == "embedding" and class_name not in ["NvidiaTextEmbedder", "NvidiaDocumentEmbedder"]:
-            # Handle the case where the model is an "embedding" and the class name is not compatible.
-            err_msg = f"Model {model.id} is an embedding, but {class_name} is not a compatible client."
-            raise ValueError(err_msg)
-        if model.model_type != "embedding" and model.client != class_name:
-            # Handle the case where the model type is not "embedding" and the client doesn't match.
-            err_msg = f"Model {model.id} is incompatible with client {class_name}. \
-                        Please check `{class_name}.available_models`."
+        elif model.model_type == "embedding" and client in ["NvidiaTextEmbedder", "NvidiaDocumentEmbedder"]:
```

Comment:
you can eliminate this check - and just check for invalid states and raise a `ValueError`